### PR TITLE
Display upselling dialog when import error is size type and user can upgrade

### LINKED ISSUE #2629: Design error modal window with upselling CTAca
When raising an error relates with your account limitationa we should show a nice CTA on the modal window for upselling

### LINKED ISSUE #2833: Going to page 1 in the public dashboard goes to the org dashboard, and not the one for the user.
https://team.cartodb.com/u/saleiva/page/2 try to go to page 1

### LINKED ISSUE #2916: Connecting a data library datasets from the datasets view still creates a new map.

### PR DESCRIPTION
**Pull request by xavijam**
Basically if user tries to import a file that doesn't fit in his/her quota, an upselling message/element should be displayed.

Preview:

screen shot 2015-03-25 at 10 31 54

---

#### When could it be displayed?
- When creates a new map (importing data library).
- Adding a new file (drag and drop).
- Importing a file that takes more space than expected.

---

#### Things to comment
- Upload model (JS) validation will not only raise a message. It could send an error code as we do in backend part. For example, 8001 error is when user can import a file that doesn't fit in his/her quota.

Fixes #2629

---
##### Apart from that...

Fixes #2833 
Fixes #2916

cc @saleiva @carlostallon @viddo

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2520/))

**Comment by viddo** [comment]
Hmm, only font changes? No JS changes then?

**Comment by xavijam** [comment]
I’m adding them! Gimme some minutes :DD


On 25 Mar 2015 at 10:58:46, Nicklas Gummesson (notifications@github.com) wrote:

Hmm, only font changes? No JS changes then?

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2524/))

**Comment by viddo** [comment]
ok, ping me again when ready to review :)

**Comment by xavijam** [comment]
Yep, thanks buddy

**Comment by xavijam** [comment]
Yep, thanks buddy

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2535/))

**Comment by saleiva** [comment]
The text is a bit recursive. On the blue box I'd write...
"Keep your data and get some more quota by upgrading your plan". 

Do you have controled the case of the user having the free trial still available?

**Comment by xavijam** [comment]
Yes, it only appears if the user can "use it".

**Comment by xavijam** [comment]
@saleiva, why that text? Take into account that "block" will appear when user is importing a file and the process is stopped due to an error, and that error is he/she doesn't have enough quota.

**Comment by saleiva** [comment]
Because if you red the title, and then the subtitle the actual text sounds contrary.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2539/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2541/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2544/))

**Comment by xavijam** [comment]
@viddo could you take a look please?

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2560/))

File: lib/assets/javascripts/cartodb/new_common/dialogs/create/create_loading.js [hunk 1]
```diff
@@ -38,6 +40,19 @@ module.exports = cdb.core.View.extend({
       d.err.item_queue_id = currentImport.get('id');
     }
 
+    if (d.state === "error") {
+      var sizeError = d.err && d.err.error_code && d.err.error_code == "8001";
+      var userCanUpgrade = !cdb.config.get('custom_com_hosted') && (!this.user.isInsideOrg() || this.user.isOrgAdmin());
```
Comment (viddo) [inline lib/assets/javascripts/cartodb/new_common/dialogs/create/create_loading.js]: This logic to decide whether to show the upgrade url is duplicated in some places by now (e.g. [create footer](https://github.com/CartoDB/cartodb/blob/d6cff88d929c5496ea0ae334f1bbfd4b53b8ecba/lib/assets/javascripts/cartodb/new_common/dialogs/create/create_footer.js#L31), [settings dropdown](https://github.com/CartoDB/cartodb/blob/dba64cee0aa5d0304d0dd021f897b1d7dcf9918c/lib/assets/javascripts/cartodb/new_common/views/dashboard_header/settings_dropdown_view.js#L59), [import](https://github.com/CartoDB/cartodb/blob/f42da41bcbbbd6d3cb02965a344c350afa963904/lib/assets/javascripts/cartodb/new_common/dialogs/create/listing/imports/import_selected_dataset_view.js#L76), possibly more), perhaps makes sense to extract it to some common place to not duplicate this if-checks every time (e.g. `cdb.config.upgradeUrl()`)?

Btw, @javisantana, right now the upgrade_url is set on the window object, should it be moved to the frontend_config (similar to the case we discussed yesterday)?

**Comment by viddo** [comment]
:+1:

File: lib/assets/javascripts/cartodb/new_common/dialogs/create/create_loading.js (hunk 1 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb/new_common/dialogs/create/create_loading.js]: Agree, wouldn't you move it to user model?

Comment (xavijam) [inline lib/assets/javascripts/cartodb/new_common/dialogs/create/create_loading.js]: After talking with @javisantana, we should do it, but in this pull request. Let's keep as it is and revisit again when needed.

**Pull request merged**